src/videojs-hls.js

```diff
@@ -679,7 +718,7 @@ videojs.Hls.prototype.fillBuffer = function(offset) {
   // being buffering so we don't preload data that will never be
   // played
   if (!this.playlists.media().endList &&
-      !this.player().hasClass('vjs-has-started') &&
+      !player.hasClass('vjs-has-started') &&
```

Comment:
Most places are checking `hasPlayed_` except here. Wondering if that's intentional or, at least, not harmful.